splink/find_brs_with_comparison_counts_below_threshold.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import string
+from typing import TYPE_CHECKING, Dict, List, Set
+
+import pandas as pd
+
+from .input_column import InputColumn
+
+if TYPE_CHECKING:
+    from .linker import Linker
+logger = logging.getLogger(__name__)
+
+
+def sanitise_column_name(column_name):
```

Comment:
This function may want to live inside [misc.py](https://github.com/moj-analytical-services/splink/blob/find_blocking_rules_that_create_comparisons_below_threshold/splink/misc.py).

It feels like something we'd want to use elsewhere.

Reply:
I am guessing this causes issues for users that have "spacey column names"?

i.e. `my column` is a valid column name in the rest of our code base, but would have the space stripped here.

Reply:
I will name things better here.

The sanitisation here is purely to help establish a correspondence between one hot encoded columns `__fixed__surname` and the blocking rules in `blocking_columns`.  I sanitise so I don't need to worry that the one hot encoded columns end up with special characters/spaces etc.

This also means spaces don't matter because the true blocking rule is held in `splink_blocking_rule`

Reply:
To illustrate where this is used

![image](https://github.com/moj-analytical-services/splink/assets/2608005/bb1a27f5-1815-429c-b476-d138596845a2)